### PR TITLE
Chore add logs to employee app

### PR DESCRIPTION
**Pull request by Szymiks**
Resolves: https://github.com/Code-Poets/sheetstorm/issues/152

Next PR with logs, now in employee app.

### CODE REVIEW

File: employees/forms.py [hunk 1]
```diff
@@ -1,14 +1,18 @@
...
 
+logger = logging.getLogger(__name__)
+
 
 class ProjectJoinForm(forms.Form):
 
     projects = forms.ChoiceField(choices=[])
 
     def __init__(self, queryset: QuerySet, *args: Any, **kwargs: Any) -> None:
+        logger.debug(f"ProjectJoinForm return all available projects to join for user: {queryset}")
```
Comment (rwrzesien) [inline employees/forms.py]: Not needed, don't log in forms.

File: employees/serializers.py [hunk 2]
```diff
@@ -20,14 +23,15 @@ def __init__(self, **kwargs: Any) -> None:
...
-            **kwargs
+            **kwargs,
         )
         self.validators[1].message = MAX_HOURS_VALUE_VALIDATOR_MESSAGE
         self.validators[2].message = MIN_HOURS_VALUE_VALIDATOR_MESSAGE
 
     def to_internal_value(self, data: str) -> Decimal:
         if isinstance(data, str) and ":" in data:
             converted = data.replace(":", ".")
+            logger.debug(f"Data: {data} has been change to: {converted}")
```
Comment (rwrzesien) [inline employees/serializers.py]: Not needed, don't log in serializers.

File: employees/views.py [hunk 3]
```diff
@@ -22,24 +23,30 @@
...
 
+logger = logging.getLogger(__name__)
+
 
 class ReportViewSet(viewsets.ModelViewSet):
     serializer_class = ReportSerializer
     permission_classes = (permissions.IsAuthenticated,)
 
     def get_queryset(self) -> QuerySet:
+        logger.info(f"Get query set from ReportViewSet for user: {self.request.user}")
```
Comment (rwrzesien) [inline employees/views.py]: `{self.request.user.pk}`

File: employees/views.py [hunk 4]
```diff
@@ -22,24 +23,30 @@
...
         return Report.objects.filter(author=self.request.user).order_by("-date")
 
     def perform_create(self, serializer: ReportSerializer) -> None:
+        logger.info(f"Perform create method for user: {self.request.user}")
```
Comment (rwrzesien) [inline employees/views.py]: `{self.request.user.pk}`

File: employees/views.py [hunk 5]
```diff
@@ -22,24 +23,30 @@
...
         serializer.save(author=self.request.user)
 
 
 def query_as_dict(query_set: QuerySet) -> Dict[str, Any]:
+    logger.debug(f"Query set: {query_set} map to dictionary")
```
Comment (rwrzesien) [inline employees/views.py]: Not needed in this function.

File: employees/views.py [hunk 6]
```diff
@@ -52,15 +59,18 @@ class ReportList(APIView):
     permission_classes = (permissions.IsAuthenticated,)
 
     def get_queryset(self) -> QuerySet:
+        logger.info(f"User: {self.request.user} get reports queryset")
```
Comment (rwrzesien) [inline employees/views.py]: `{self.request.user.pk}`

File: employees/views.py [hunk 7]
```diff
@@ -52,15 +59,18 @@ class ReportList(APIView):
...
         return Report.objects.filter(author=self.request.user).order_by("-date", "project__name")
 
     def _add_project(self, serializer: ReportSerializer, project: Project) -> None:
+        logger.info(f"Add project method for user: {self.request.user} to project: {project}")
```
Comment (rwrzesien) [inline employees/views.py]: `{self.request.user.pk}`

File: employees/forms.py (hunk 1 above)
Comment (Szymiks) [reply employees/forms.py]: Done

File: employees/views.py (hunk 3 above)
Comment (Szymiks) [reply employees/views.py]: Done

File: employees/views.py (hunk 4 above)
Comment (Szymiks) [reply employees/views.py]: Done

File: employees/views.py (hunk 5 above)
Comment (Szymiks) [reply employees/views.py]: Done

File: employees/views.py (hunk 6 above)
Comment (Szymiks) [reply employees/views.py]: Done

File: employees/views.py (hunk 7 above)
Comment (Szymiks) [reply employees/views.py]: Done

File: employees/forms.py [hunk 8]
```diff
@@ -1,8 +1,11 @@
+import logging
 from typing import Any
 
 from django import forms
 from django.db.models import QuerySet
 
+logger = logging.getLogger(__name__)
```
Comment (rwrzesien) [inline employees/forms.py]: Is this still needed?

File: employees/views.py [hunk 9]
```diff
@@ -22,15 +23,19 @@
...
 
+logger = logging.getLogger(__name__)
+
 
 class ReportViewSet(viewsets.ModelViewSet):
     serializer_class = ReportSerializer
     permission_classes = (permissions.IsAuthenticated,)
 
     def get_queryset(self) -> QuerySet:
+        logger.info(f"Get query set from ReportViewSet for user with id: {self.request.user.pk}")
```
Comment (rwrzesien) [inline employees/views.py]: Not needed.

File: employees/views.py [hunk 10]
```diff
@@ -22,15 +23,19 @@
...
         return Report.objects.filter(author=self.request.user).order_by("-date")
 
     def perform_create(self, serializer: ReportSerializer) -> None:
+        logger.info(f"Perform create method for user with id: {self.request.user.pk}")
```
Comment (rwrzesien) [inline employees/views.py]: Not needed.

File: employees/views.py [hunk 11]
```diff
@@ -40,6 +45,7 @@ def query_as_dict(query_set: QuerySet) -> Dict[str, Any]:
         key = record.date
         dictionary.setdefault(key, [])
         dictionary[key].append(record)
+    logger.debug(f"Result of mapping. Data of dictionary: {dictionary}")
```
Comment (rwrzesien) [inline employees/views.py]: Not needed.

File: employees/views.py [hunk 12]
```diff
@@ -164,10 +188,13 @@ def post(self, request: HttpRequest, pk: int) -> Union[Response, HttpResponseRed
...
             serializer.save()
+            logger.info(f"Report with id: {report.pk} has been updated by user with id: {request.user.pk}")
         return redirect("custom-report-list")
 
 
 def delete_report(_request: HttpRequest, pk: int) -> HttpResponseRedirectBase:
+    logger.info(f"User with id: {_request.user.pk} want to delete report with id: {pk}")
     report = get_object_or_404(Report, pk=pk)
     report.delete()
+    logger.debug(f"Report with id: {report.pk} has been deleted")
```
Comment (rwrzesien) [inline employees/views.py]: `report.pk` will be `None` here after deleting so there is no need to print it.

File: employees/views.py (hunk 9 above)
Comment (Szymiks) [reply employees/views.py]: Done

File: employees/views.py (hunk 10 above)
Comment (Szymiks) [reply employees/views.py]: Done

File: employees/views.py (hunk 12 above)
Comment (Szymiks) [reply employees/views.py]: Done

File: employees/forms.py (hunk 8 above)
Comment (Szymiks) [reply employees/forms.py]: not needed :)

File: employees/views.py (hunk 12 above)
Comment (Szymiks) [reply employees/views.py]: DOne

File: employees/views.py (hunk 11 above)
Comment (Szymiks) [reply employees/views.py]: Done

File: employees/forms.py [hunk 13]
```diff
@@ -1,3 +1,4 @@
+import logging
```
Comment (rwrzesien) [inline employees/forms.py]: Same here! :)

Comment (Szymiks) [reply employees/forms.py]: Done:)

**Pull request closed without merging**